bins/revme/src/statetest/runner.rs

```diff
@@ -350,58 +363,82 @@ pub fn run(
     if trace {
         single_thread = true;
     }
+    let n_files = test_files.len();
 
     let endjob = Arc::new(AtomicBool::new(false));
-    let console_bar = Arc::new(ProgressBar::new(test_files.len() as u64));
-    let mut joins: Vec<std::thread::JoinHandle<Result<(), TestError>>> = Vec::new();
-    let queue = Arc::new(Mutex::new((0, test_files)));
+    let console_bar = Arc::new(ProgressBar::new(n_files as u64));
+    let queue = Arc::new(Mutex::new((0usize, test_files)));
     let elapsed = Arc::new(Mutex::new(std::time::Duration::ZERO));
-    let num_threads = if single_thread { 1 } else { 10 };
-    for _ in 0..num_threads {
+
+    let num_threads = match (single_thread, std::thread::available_parallelism()) {
+        (true, _) | (false, Err(_)) => 1,
+        (false, Ok(n)) => n.get(),
+    };
+    let num_threads = num_threads.min(n_files);
+    let mut handles = Vec::with_capacity(num_threads);
+    for i in 0..num_threads {
         let queue = queue.clone();
         let endjob = endjob.clone();
         let console_bar = console_bar.clone();
         let elapsed = elapsed.clone();
 
-        let mut thread = std::thread::Builder::new();
+        let mut thread = std::thread::Builder::new().name(format!("runner-{i}"));
 
         // Allow bigger stack in debug mode to prevent stack overflow errors
-        //if cfg!(debug_assertions) {
-        thread = thread.stack_size(4 * 1024 * 1024);
-        //}
-
-        joins.push(
-            thread
-                .spawn(move || loop {
-                    let (index, test_path) = {
-                        let mut queue = queue.lock().unwrap();
-                        if queue.1.len() <= queue.0 {
-                            return Ok(());
-                        }
-                        let test_path = queue.1[queue.0].clone();
-                        queue.0 += 1;
-                        (queue.0 - 1, test_path)
-                    };
-                    if endjob.load(Ordering::SeqCst) {
-                        return Ok(());
-                    }
-                    //println!("Test:{:?}\n",test_path);
-                    if let Err(err) = execute_test_suit(&test_path, &elapsed, trace) {
-                        endjob.store(true, Ordering::SeqCst);
-                        println!("Test[{index}] named:\n{test_path:?} failed: {err}\n");
-                        return Err(err);
-                    }
+        if cfg!(debug_assertions) {
+            thread = thread.stack_size(4 * 1024 * 1024);
+        }
```

Comment:
Stack size got increased and default release stack size was not enough. Lets see if tests are going to pass if they overflow revert tihs.